server/src/internalClusterTest/java/org/opensearch/search/backpressure/SearchBackpressureIT.java

```diff
@@ -0,0 +1,268 @@
+                latch.countDown();
+            }
+
+            @Override
+            public void onFailure(Exception e) {
+                assertEquals(TaskCancelledException.class, e.getClass());
+                assertTrue(e.getMessage().contains("elapsed time exceeded"));
+                latch.countDown();
+            }
+        });
+
+        latch.await(TIMEOUT.getSeconds(), TimeUnit.SECONDS);
+    }
+
+    public void testSearchShardTaskCancellationWithHighCpu() throws InterruptedException {
+        Settings request = Settings.builder()
+            .put(SearchBackpressureSettings.SETTING_MODE.getKey(), "enforced")
+            .put(CpuUsageTracker.SETTING_CPU_TIME_MILLIS_THRESHOLD.getKey(), 15000)
+            .build();
+        assertAcked(client().admin().cluster().prepareUpdateSettings().setPersistentSettings(request).get());
+
+        CountDownLatch latch = new CountDownLatch(1);
+
+        client().execute(TestTransportAction.ACTION, new TestRequest(RequestType.HIGH_CPU), new ActionListener<>() {
+            @Override
+            public void onResponse(TestResponse testResponse) {
+                fail("SearchShardTask should have been cancelled");
+                latch.countDown();
+            }
+
+            @Override
+            public void onFailure(Exception e) {
+                assertEquals(TaskCancelledException.class, e.getClass());
+                assertTrue(e.getMessage().contains("cpu usage exceeded"));
+                latch.countDown();
+            }
+        });
+
+        latch.await(TIMEOUT.getSeconds(), TimeUnit.SECONDS);
+    }
+
+    public void testSearchCancellationWithBackpressureDisabled() throws InterruptedException {
+        Settings request = Settings.builder().put(SearchBackpressureSettings.SETTING_MODE.getKey(), "monitor_only").build();
+        assertAcked(client().admin().cluster().prepareUpdateSettings().setPersistentSettings(request).get());
+
+        CountDownLatch latch = new CountDownLatch(1);
+
+        client().execute(TestTransportAction.ACTION, new TestRequest(RequestType.HIGH_ELAPSED_TIME), new ActionListener<>() {
+            @Override
+            public void onResponse(TestResponse testResponse) {
+                latch.countDown();
+            }
+
+            @Override
+            public void onFailure(Exception e) {
+                fail("SearchShardTask shouldn't have cancelled for monitor_only mode");
+                latch.countDown();
+            }
+        });
+
+        latch.await(TIMEOUT.getSeconds(), TimeUnit.SECONDS);
+    }
+
+    enum RequestType {
+        HIGH_CPU,
+        HIGH_ELAPSED_TIME;
+    }
+
+    public static class TestRequest extends ActionRequest {
+        private final RequestType type;
+
+        public TestRequest(RequestType type) {
+            this.type = type;
+        }
+
+        public TestRequest(StreamInput in) throws IOException {
+            super(in);
+            this.type = in.readEnum(RequestType.class);
```

Comment:
Please override the corresponding `writeTo` method in this class so that the RequestType is written to the stream.

Reply:
Got it. Overriding `writeTo` method in the next commit.